### PR TITLE
Open notebook in JupyterLab

### PR DESCRIPTION
**Pull request by elizzhao**
Can now open a notebook in JupyterLab by selecting the option to do so in the 3dot menu on the notebook card. 

Note: Waiting on more info/understanding of JupyterLab's autosaving semantics from callisto before proceeding

### CODE REVIEW

File: src/pages/workspaces/workspace/notebooks/NotebookLauncher.js
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/notebooks/NotebookLauncher.js]:
```suggestion
        iframeChoice === 'lab' && linkButton({
```

**Review by zarsky-broad — APPROVED** [review]
Thinking about it, we might use same prop for specifying rstudio or something, so maybe just call it `app` instead of `iframeChoice`?

**Comment by elizzhao** [comment]
NOTE: this PR is being put on hold until we are able to properly change juypterlab's autosave to match how we do notebooks. Currently lab only autosaves every 2min which is a) slow and b) different from regular notebooks where our autosaving is more complete

File: src/pages/workspaces/workspace/notebooks/NotebookLauncher.js
Comment (elizzhao) [reply src/pages/workspaces/workspace/notebooks/NotebookLauncher.js]: added! thanks

**Review by katevoss — COMMENTED** [review]
I know that this is on hold, but when I clicked the "Open with JupyterLab" link, I got a 404 page. I was able to open the notebook normally.
screen shot 2018-12-18 at 1 57 01 pm

**Comment by zarsky-broad** [comment]
@katevoss yes, probably because you don't have lab on your cluster and this doesn't currently include a cluster version bump.

**Comment by elizzhao** [comment]
Notification added that lets the user know the autosave interval is 2min. Would love feedback on the messaging! Notification is up for 30s and is dismissible. 
screen shot 2019-01-14 at 3 07 11 pm

File: src/pages/workspaces/workspace/notebooks/NotebookLauncher.js
Comment (katevoss) [inline src/pages/workspaces/workspace/notebooks/NotebookLauncher.js]: Is there a way that users can force-save their notebook before exiting? Can they save their notebook in the menu or with a keyboard shortcut? If so then I'd recommend explicitly telling them to Save your notebook before closing this notebook or window, or something like that.

Comment (elizzhao) [reply src/pages/workspaces/workspace/notebooks/NotebookLauncher.js]: Yeah, you can command+S or press the save button to force save. I'll change the messaging to just tell them to save before exiting

**Comment by bradtaylor** [comment]
@elizzhao Please do not merge this until after the nurses workshop (so, not until next Monday)

Message looks good to me

**Comment by breilly2** [comment]
I just learned from Rob that Leo very recently (yesterday?) added support for JupyterLab plugins which may allow us to tweak the autosave interval.

**Comment by elizzhao** [comment]
Yeah, Adrian let me know that extensions came out like yesterday or today. I can throw a ticket on the board specifically for extensions, but since this has been sitting for a while, I think it might be best to push this through, and then do the extensions separately.

File: src/pages/workspaces/workspace/notebooks/NotebookLauncher.js
Comment (bradtaylor) [inline src/pages/workspaces/workspace/notebooks/NotebookLauncher.js:239]: I would add "by clicking the save icon"

Also:
"JupyterLab is new in Terra. We are working to improve its integration. Please contact us with any questions or feedback you may have."

**Pull request merged**
